readers/api/transport.go

```diff
@@ -0,0 +1,150 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+	"time"
+
+	kithttp "github.com/go-kit/kit/transport/http"
+	"github.com/go-zoo/bone"
+	"github.com/mainflux/mainflux"
+	"github.com/mainflux/mainflux/readers"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	contentType = "application/json"
+	defLimit    = 10
+	defOffset   = 0
+)
+
+var (
+	errInvalidRequest     = errors.New("received invalid request")
+	errUnauthorizedAccess = errors.New("missing or invalid credentials provided")
+	auth                  mainflux.ThingsServiceClient
+)
+
+// MakeHandler returns a HTTP handler for API endpoints.
+func MakeHandler(svc readers.MessageRepository, tc mainflux.ThingsServiceClient, svcName string) http.Handler {
+	auth = tc
+
+	opts := []kithttp.ServerOption{
+		kithttp.ServerErrorEncoder(encodeError),
+	}
+
+	mux := bone.New()
+	mux.Get("/channels/:chanID/messages", kithttp.NewServer(
+		listMessagesEndpoint(svc),
+		decodeList,
+		encodeResponse,
+		opts...,
+	))
+
+	mux.GetFunc("/version", mainflux.Version(svcName))
+	mux.Handle("/metrics", promhttp.Handler())
+
+	return mux
+}
+
+func decodeList(_ context.Context, r *http.Request) (interface{}, error) {
+	chanID, err := strconv.ParseUint(bone.GetValue(r, "chanID"), 10, 64)
+	if err != nil {
+		return nil, errInvalidRequest
+	}
+
+	if err := authorize(r, chanID); err != nil {
+		return nil, err
+	}
+
+	offset, err := getQuery(r, "offset", defOffset)
+	if err != nil {
+		return nil, err
+	}
+
+	limit, err := getQuery(r, "limit", defLimit)
+	if err != nil {
+		return nil, err
+	}
+
+	req := listMessagesReq{
+		chanID: chanID,
+		offset: offset,
+		limit:  limit,
+	}
+
+	return req, nil
+}
+
+func encodeResponse(_ context.Context, w http.ResponseWriter, response interface{}) error {
+	w.Header().Set("Content-Type", contentType)
+
+	if ar, ok := response.(mainflux.Response); ok {
```

Comment:
Rename `ar`.

Reply:
This variable is called like this everywhere, so I don't see a reason for that. If you want to change it, send separate PR where you will update it in other services too.